tests/ExampleApps/PickingDemoApp/main.cpp

```diff
@@ -0,0 +1,153 @@
+            std::cout << "Selected objects (" << pickResults.size() << ") : " << std::endl;
+            for ( const auto& pr : pickResults )
+            {
+                std::cout << pr.m_roIdx << ' ';
+            }
+            std::cout << std::endl;
+            std::cout << "Selection buffers  (" << pickResults.size() << ") : " << std::endl;
+            for ( const auto& pr : pickResults )
+            {
+                processPickResult( pr );
+            }
+            std::cout << std::endl;
+        }
+        else
+        {
+            std::cout << "Single selection mode : " << std::endl;
+            // Access only the pixel from the selection buffer below the mouse
+            processPickResult( pickingResult );
+        }
+    }
+
+    void processPickResult( const Ra::Engine::Rendering::Renderer::PickingResult& pickingResult ) {
+
+        //! [processing the picking info]
+        std::cout << "Picking mode          : " << pickingResult.m_mode << std::endl;
+        std::cout << "RenderObject selected : " << pickingResult.m_roIdx << std::endl;
+        std::cout << "Pixels in the selection : " << pickingResult.indices().size() << std::endl;
+        pickingResult.removeDuplicatedIndices();
+        std::cout << "Nearest [face, vertex, edge] by pixels : " << std::endl;
+        for ( const auto& [f, v, e] : pickingResult.indices() )
+            std::cout << f << ' ' << v << ' ' << e << '\n';
+    }
+
+  private:
+    bool m_isTracking {false};
+};
+#include "main.moc"
+
+/**
+ *      Main code to demonstrate how to manage selection
+ */
+
+/**
+ * Define a factory that set the wanted renderer on the window
+ */
+class DemoWindowFactory : public Ra::Gui::BaseApplication::WindowFactory
+{
+  public:
+    ~DemoWindowFactory() = default;
+    inline Ra::Gui::MainWindowInterface* createMainWindow() const override {
+        auto window = new DemoWindow();
+        return window;
+    }
+};
+
+void addDummyScene( Ra::Gui::BaseApplication& app ) {
+    auto cube = Ra::Core::Geometry::makeSharpBox( {0.1f, 0.1f, 0.1f} );
+    auto e    = app.m_engine->getEntityManager()->createEntity( "Cube" );
+    auto c =
+        new Ra::Engine::Scene::TriangleMeshComponent( "Cube Mesh", e, std::move( cube ), nullptr );
+
+    auto geometrySystem = app.m_engine->getSystem( "GeometrySystem" );
+    geometrySystem->addComponent( e, c );
+    app.m_mainWindow->prepareDisplay();
+}
+
+/**
+ * main function.
+ */
+int main( int argc, char* argv[] ) {
+
+    //! [Instatiating the application]
+    Ra::Gui::BaseApplication app( argc, argv );
+    //! [Instatiating the application]
+
+    //! [Initializing the application]
+    // The customization functor is called here
+    app.initialize( DemoWindowFactory() );
+    //! [Initializing the application]
+
+    if ( argc < 2 )
+    {
+        // Create a scene when called without input file
+        addDummyScene( app );
+    }
```

Comment:
Is there any else ? I have tried if something magic here, but giving an arg to the app do not display anything.

Reply:
See base application options. You need -f to load a scene. My check is a bit hacky, true.